pkg/gator/runner.go

```diff
@@ -282,6 +283,27 @@ func (r *Runner) runReview(ctx context.Context, newClient func() (Client, error)
 		}
 	}
 
+	// check to see if obj is an AdmissionReview kind
+	if toReview.GetKind() == "AdmissionReview" && toReview.GroupVersionKind().Group == admissionv1.SchemeGroupVersion.Group {
+		// convert unstructured into AdmissionReview, don't allow unknown fields
+		var ar admissionv1.AdmissionReview
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructuredWithValidation(toReview.UnstructuredContent(), &ar, true); err != nil {
+			return nil, fmt.Errorf("%w: unable to convert to an AdmissionReview object, error: %v", ErrInvalidK8sAdmissionReview, err)
+		}
+
+		if ar.Request == nil { // then this admission review did not actually pass in an AdmissionRequest
+			return nil, fmt.Errorf("%w: request did not actually pass in an AdmissionRequest", ErrMissingK8sAdmissionRequest)
+		}
+
+		arr := target.AugmentedReview{
```

Comment:
Do we also need to replicate this webhook logic that rewrites Object for DELETE?

https://github.com/open-policy-agent/gatekeeper/blob/91d29f6c9cc7a41bc0a8b766e4b6a138a529777f/pkg/webhook/policy.go#L150-L165

Reply:
I went back and forth here on enforcing the `oldObject` nullability on `DELETE` ops. But I think it makes sense to make this `gator verify` track the g8r behavior as closely as possible, which means mimicking the k8s api here.

Reply:
One aspect I tried was to dedup the code but it wasn't immediately obvious to me how to do so without pulling in more dependencies (like `admission.Denied` resp into the gator `runReview` flow). I added a TODO to revisit things here eventually.

Reply:
Yeah, probably the answer would be to move the admission request processing code into it's own package (or this one depending on how the import cycles look and if it makes sense from a package organization perspective), then import/call.

Reply:
if we embarked on a refactor mayhaps we could pull in some of those as well:

https://github.com/open-policy-agent/gatekeeper/issues/1779